### ISSUE TITLE
Potential issues with an Everdrive Cart

### ISSUE THREAD

**Issue by ShawMerlin** [ShawMerlin/NES-Slotmaster#5]
We have had a few reports of issues when using an Everdrive Cartridge.

This is being looked at currently to see if a root cause can be found.
It might be because the ribbon cable is long and there is more distance than the stock connector.

I am opening this issue in case others want to share their experience on here and to keep updated on our progress.

Thanks,
-Merlin Shaw

**Comment by dbstallman1** [comment]
Here has been my experience - I purchased three Slotmasters from LaserBear.  I installed the first one and my EverDrive N8 with latest firmware appeared to work until a game was played.  It would then lock up or have graphical glitches within a minute or two.  I installed the second one the same night and the Everdrive seemed to work correctly.  

Then next day I installed the third and that system would not recognize the Everdrive cart at all, and when going back to the other two systems they no longer would boot either.  I saw on the product page that the 8 pin header was needed for proper operation, so I ordered the headers and interconnects.  When they arrived I added them and tried again.

The first system still boots the Everdrive, but gives alternating error codes - SD card initialization failed or SD card not formatted.  I will add that this Everdrive continues to work perfectly in my top loader.  The second and third systems will now do the blinking light and reset.

All three systems work with any other cart (beyond Dragon Warrior) that I put in them.  The success with the Everdrive has been failure far more often then working.

**Comment by ShawMerlin** [comment]
I think the issue you are experiencing is due to an earlier Gerber file that had thinner traces.  This results in not enough power getting into the Everdrive Cartridge.

If you are able, bodge the power pins from the 72 pin to the 64 pin connector so that it will provide more current.
From looking at the back of the boards, it will be the 2 pins on the left and the bottom pin on the far right.

Let us know if that solves this issue.

Here is a picture of the Power Traces, they are twice as thick.  check your boards to see what size they are compared to the other traces.
<img width="889" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/ac232006-795a-4ae1-a355-00568ce9d9c4">

<img width="602" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/f904b488-fd5d-4b62-830a-764397502cef">

**Comment by dbstallman1** [comment]
From a cursory glance they appear to be the same size.  I have v1.8 of the boards.

I'm heading out for an hour or so right now, but I can definitely get this done on at least one of them tonight. Reinforcing the connections on both sets of PCBs is what is best, correct?

**Comment by ShawMerlin** [comment]
Yes, use a wire between the thicker trace points above on the 72 pin and connect to the related pad on the 64 pin.  By reinforcing with a wire, this will allow for the higher current for the Everdrive to pass through and should resolve your issue.

**Comment by dbstallman1** [comment]
I've added reinforcement jumpers to all three pins on the top and bottom boards on one of sets.  Unfortunately, it didn't make a difference.  All other carts still work.  I'll do the mod to the other two tomorrow night to see if that makes a difference on them.

So far, though, no success.

**Comment by ShawMerlin** [comment]
I did explain one of the top pins wrong and I believe you reinforced the System CLK and not the GND Pin.  Here is a diagram showing the top board from the front.  On the left, the top pin should be reinforced to the top pin on the IDC connector. 

<img width="1130" alt="image" src="https://github.com/ShawMerlin/NES-Slotmaster/assets/70423454/2565b768-e852-4934-b11f-6f0aec38d40c">

**Comment by dbstallman1** [comment]
I actually couldn't remember which side was only one pin once I was at the workbench.  Rather than go back and check, I reinforced pins 1, 36, 37, and 72.  So that means I did the correct ones anyway.

**Comment by ShawMerlin** [comment]
Thank you for following up with that.  I will see if I can get to the bottom of this issue.  If you can send me an email  merlin (dot) shaw at google.  I will make sure to send you a board to test.

**Comment by ShawMerlin** [comment]
At this point.  We feel that it's a Noise/EMI issue with the Data lines becoming corrupted.  This could be due to the longer cable but we are seeing it as well with a shorter cable.

Some Everdrives work fine with no issue so it's very hit or miss.  We are working on a solution that will improve compatiblity with Everdrive Carts.  Our goal is to have the same compatibility as an Everdrive with the original connector.

I will update this thread when we have more news on this.

**Comment by ShawMerlin** [comment]
Thank you everyone for the support!!  We have our new solution here and I will be sending it to everyone that was willing to test for us.  If you have not already, please get me an email with your address so I can send you one - 

Here is the IDC Cable Replacement
https://github.com/ShawMerlin/NES-Slotmaster/tree/main/NES_Ampersand

**Comment by Tzibu** [comment]
***@***.*** ***@***.***>  please 😊

 

Lähettäjä: Merlin Shaw ***@***.***> 
Lähetetty: lauantai 15. kesäkuuta 2024 19.20
Vastaanottaja: ShawMerlin/NES-Slotmaster ***@***.***>
Kopio: Subscribed ***@***.***>
Aihe: Re: [ShawMerlin/NES-Slotmaster] Potential issues with an Everdrive Cart (Issue #5)

 

Thank you everyone for the support!! We have our new solution here and I will be sending it to everyone that was willing to test for us. If you have not already, please get me an email with your address so I can send you one -

Here is the IDC Cable Replacement
https://github.com/ShawMerlin/NES-Slotmaster/tree/main/NES_Ampersand

**Comment by ShawMerlin** [comment]
It masks your email address, you will have to email me.  I'm sure if you
look hard enough, you can find my email ;)

**Comment by otacon6530** [comment]
> Hello,
> Is this still an issue? I recently built for myself a Slotmaster (rev. 1.8D) and connected the two boards with a NES Ampersand.
> 
> I had no issue with my (clone) Everdrive N8, or my original games, but I am having the issues described here with a (clone) Everdrive N8 Pro/Plus: The cart appears to boot fine to the menu, can start games, but will show graphical corruption/glitches and/or will freeze after a short while.
> 
> I don't seem to find a workaround mentioned anywhere.

Could it be short pinned or banded?
